### DISCUSSION TITLE
Stage Display purpose per prim?

### DISCUSSION THREAD

**Discussion by daviddeacon** [Autodesk/maya-usd#1565] [Q&A]
Is it possible to have parts of a stage be displayed at different quality levels? 

For example, I'm testing out possible environment build workflows, and have a number of trees in a stage, and the stage set to display proxy purpose. I would like to be able to select one or more trees that are close to the camera and set their display purpose to render, to be able to see the final result more accurately. I think my only current option is to change the display purpose of the entire stage. Which can tank performance on larger scenes.

If there is a work around please let me know, if not can this be added as a feature request? 

David.

**Comment by williamkrick** [comment]
Which version of Maya and MayaUSD are you working with? If you have Ufe v2 available then Purpose should show up in the Attribute Editor and you can manually switch the purpose of the interesting trees to Proxy so they show up. You'd also have to switch the purpose of the actual Proxy trees to something else so they don't draw. And you'd have to remember to clear all that state before finishing.

**Reply by daviddeacon** [reply]
Thanks, I'm using Maya 2022 so your method does work, but I fear that getting artists to 'remember' to change the settings back might be asking for trouble... 

In the future if it's possible to have prims displayed at different quality levels, using 1,2,3 keyboard shortcuts like when switching on subdivision preview might be a nice workflow? eg 1=guide,2=proxy,3=render. Especially if it could work on the hierarchy, again just like subdivision preview.

David.
